src/ConsistencyAndReplicationModel.md

```diff
@@ -58,3 +57,14 @@ In the following failure case, exactly-once guarantee can be broken:
 In the following failure case, invocation state will become indeterminate:
 
 * As explained above, when an invocation does not receive a response in time, invocation will fail with an `OperationTimeoutException`. This exception does not say anything about outcome of the operation, that means operation may not be executed at all, it may be executed once or twice (due to member left case explained above).
+
+## IndeterminateOperationStateException
+
+As described in [Invocation Lifecycle](#invocation-lifecycle) section, for partition-based invocations, such as `map.put()`, a caller waits with a timeout for the operation is executed on corresponding partition's primary replica and backup replicas, based on the sync backup configuration of the distributed data structure. Hazelcast 3.9 introduces a new mechanism to detect indeterminate situations while making such invocations. If `hazelcast.operation.fail.on.indeterminate.state` system property is enabled, an invocation throws `IndeterminateOperationStateException` when it encounters following cases:
```

Comment:
.....with a timeout for the operation that is executed on.............when it encounters the following cases: